corehq/apps/custom_data_fields/dbaccessors.py

```diff
@@ -0,0 +1,14 @@
+def get_by_domain_and_type(domain, field_type):
+    from corehq.apps.custom_data_fields.models import CustomDataFieldsDefinition
+
+    return CustomDataFieldsDefinition.get_db().view(
+        'custom_data_fields/by_field_type',
+        key=[domain, field_type],
+        include_docs=True,
+        reduce=False,
+        # if there's more than one,
+        # it's probably because a few were created at the same time
+        # due to a race condition
+        # todo: a better solution might be to use locking in this code
+        limit=1,
+    ).one()
```

Comment:
might be easier for callers of this function if you return ['doc'] though there may be some edge cases to test for if it doesn't return anything